runtime/datastructures/src/decls.rs

```diff
@@ -69,6 +101,86 @@ impl Hash for KElem {
   }
 }
 
+fn hash_map_compare<K,V>(a: &HashMap<K, V>, b: &HashMap<K, V>) -> Ordering
```

Comment:
This is not a good approach. You should modify the library to use OrdMap and OrdSet and then just make use of their Ord implementation

Reply:
I assume you mean BTreeMap, which doesn't implement ORD. It only implements PartialOrd, because there is no total ordering on maps other than the one I forced here. So even if we use BTreeMap I would need to write nearly the same code (just no need for the BinaryHeap).

Reply:
Answered: im::OrdMap